sympy/tensor/array/tests/test_array_comprehension.py

```diff
@@ -0,0 +1,14 @@
+from sympy.tensor.array.array_comprehension import ArrayComprehension
+from sympy.abc import i, j
+
+
+def test_array_comprehension():
+    a = ArrayComprehension(i*j, (i, 1, 3), (j, 2, 4))
+    b = ArrayComprehension(i, (i, 1, j))
+    c = ArrayComprehension(i*j, (i, 1, 3))
+    assert a.doit() == [[2, 3, 4], [4, 6, 8], [6, 9, 12]]
```

Comment:
can you also add `assert isinstance(a.doit(), ImmutableDenseNDimArray)`?

Reply:
Do does it mean that ArrayComprehension should inherit from ImmutableDenseNDimArray?